### PR TITLE
feat: retain selected tab when selecting deployments

### PR DESCRIPTION
**Pull request by Ash-exp**
# Description

Helm Apps > Deployment history > Retain the selected tab(eg. values.yaml) when selecting deployments. Also aligned centre the EmptyState component.

Fixes [AB 1998](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=1998)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- Go to any Helm Apps > Deployment history > Select any tab (eg. values.yaml)
- The selected tab should persist for version switch. 
- For any selected tab the EmptyState component should aligned centre.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

https://user-images.githubusercontent.com/68757539/218400622-1faa17d1-7736-4ba5-a8dd-777d085a5026.mov

Screenshot 2023-02-13 at 1 22 46 PM

### CODE REVIEW

File: src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx [hunk 1]
```diff
@@ -114,7 +114,7 @@ function ChartDeploymentHistory({
         setSelectedDeploymentHistoryIndex(index)
 
         // Resetting the deployment tab selection, loading & error states on version change.
-        setSelectedDeploymentTabIndex(0)
+        // setSelectedDeploymentTabIndex(0)
```
Comment (shivani170) [inline src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx:117]: Please remove the commented code

File: src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx [hunk 2]
```diff
@@ -314,9 +314,11 @@ function ChartDeploymentHistory({
                 (selectedDeploymentTabIndex === 2 && !selectedDeploymentManifestDetail.manifest))
         ) {
             return (
-                <CDEmptyState
-                    subtitle={`${deploymentTabs[selectedDeploymentTabIndex]} is not available for this deployment`}
-                />
+                <div className='flex h-100'>
+                    <CDEmptyState
+                        subtitle={`${deploymentTabs[selectedDeploymentTabIndex]} is not available for this deployment`}
```
Comment (shivani170) [inline src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx:319]: Please move the hardcoded text in the constant messaging > `ERROR_EMPTY_SCREEN`

File: src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx (hunk 1 above)
Comment (Ash-exp) [reply src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx]: updated 👍

File: src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx (hunk 2 above)
Comment (Ash-exp) [reply src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx]: updated 👍

**Review by shivani170 — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**